src/smart-components/common/edit-approval-workflow.js

```diff
@@ -56,43 +56,52 @@ const EditApprovalWorkflow = ({
     ).then(() => stateDispatch({ type: 'setFetching', payload: false }));
   }, []);
 
-  const onSubmit = (values) => {
+  const onSubmit = (formData) => {
     history.push(pushParam);
-    const approvalWorkflow = data ? data[0] : undefined;
+    const toUnlinkWorkflows = data.filter(
```

Comment:
I think it might be easier to use initialValue of the field from the second argument you get to the onSubmit function, to determine the linked/unlinked workflows.
```jsx
const onSubmit = (formData, formApi) => {
  formApi.getState().initialValues.fieldName
  ...
```